### ISSUE TITLE
Errors when launching games with Gamescope

### ISSUE THREAD

**Issue by troy0h** [ValveSoftware/gamescope#1346]
The two games I currently use Gamescope on both have errors when launching.
Helldivers 2: (gamescope -f -w 2560 -h 1440 -- mangohud %command% --use-d3d11)
Team Fortress 2: (gamescope -f -w 2560 -h 1440 -- mangohud %command%)
Both games worked perfectly in Gamescope fairly recently, and now crash very soon after getting into the game.

CreateSwapchainHKR: Creating swapchain for non-Gamescope swapchain.
Hooking has failed somewhere!
You may have a bad Vulkan layer interfering.

If I press OK, I get another error:

QueuePresentKHR: Attempting to present a non-hooked swapchain.
Hooking has failed somewhere!
You may have a bad Vulkan layer interfering.

**Comment by sharkautarch** [comment]
@troy0h do you also have that issue when running w/o mangohud?

**Comment by Joshua-Ashton** [comment]
Yes, it seems the mangohud layer is broken here. You can just workaround it by using --mangoapp instead,

**Comment by matte-schwartz** [comment]
I've seen this error as well recently, on my NVIDIA rig with driver 555.42.02. For example, I'm able to trigger the Vulkan layer pop-up by using the Performance Overlay within this nested steam window:
`gamescope -e -f -H 1440 -W 3440 --hdr-enabled -- steam -steamdeck -steampal -tenfoot -steamos3`

Toggling the performance overlay on and trying to launch Ori and the Will of the Wisps gives me:
```
QueuePresentKHR: Attempting to present a non-hooked swapchain.
Hooking has failed somewhere!
You may have a bad Vulkan layer interfering.
```
while keeping it off lets the game launch perfectly with HDR enabled.

I just tried compiling MangoHud from source and I seem to be able to use `gamescope -e -f -H 1440 -W 3440 --hdr-enabled --mangoapp -- steam -steamdeck -steampal -tenfoot -steamos3` to add the MangoHud overlay without swapchain warnings now.

**Comment by troy0h** [comment]
> Yes, it seems the mangohud layer is broken here. You can just workaround it by using --mangoapp instead,

Yep, this fixed it, seems like it was just a mangohud issue

**Comment by Virkkunen** [comment]
Facing the same issue recently. Running with `--mangoapp` from my Arch installation will open the game but without mangohud, almost as if the command is being ignored.

**Comment by Joshua-Ashton** [comment]
Let me guess, your distro does not build mangoapp? :)

**Comment by troy0h** [comment]
> Facing the same issue recently. Running with `--mangoapp` from my Arch installation will open the game but without mangohud, almost as if the command is being ignored.

weird, im on arch too and --mangoapp works fine for me
my launch options for helldivers 2 right now is
`gamescope -f -r 165 -w 2560 -h 1440 --mangoapp -- %command% --use-d3d11`
and that works fine for me, I don't have a mangoapp package
maybe you forgot to install the `lib32-mangohud` package?

**Comment by Virkkunen** [comment]
`lib32-mangohud` and `mangohud` are both installed from pacman, I'm able to run `mangoapp` on the terminal and it gives me an error about the session not being initialised, but it shows mangoapp exists.

**Comment by sharkautarch** [comment]
> `lib32-mangohud` and `mangohud` are both installed from pacman, I'm able to run `mangoapp` on the terminal and it gives me an error about the session not being initialised, but it shows mangoapp exists.


Maybe try reinstalling mangohud w/:
`pacman -S --overwrite='*' mangohud lib32-mangohud`

**Comment by troy0h** [comment]
> > Maybe try reinstalling mangohud w/:
> > pacman -S --overwrite='*' mangohud lib32-mangohud
> 
> Well, this did something, now running with `mangoapp` it'll show up the mangohud overlay and a black screen, a few seconds later the same two warnings will pop up

That could be the HDR stuff, maybe try removing that and see what happens

**Comment by Joshua-Ashton** [comment]
Do you have other Vulkan layers interfering like it asks? =)

**Comment by troy0h** [comment]
> Do you have other Vulkan layers interfering like it asks? =)

yeah thats what i was thinking with the HDR commands

**Comment by Joshua-Ashton** [comment]
What HDR commands? The Gamescope args should be fine

**Comment by troy0h** [comment]
> What HDR commands? The Gamescope args should be fine

the `--hdr-enabled --hdr-itm-enable --hdr-itm-sdr-nits 300 --hdr-itm-target-nits 1000` in his gamescope args, it can cause issues iirc, especially with mangohud

**Comment by Virkkunen** [comment]
Found out that the `nvidia-open-beta-dkms` (driver 555) was the issue. Changing to `nvidia-beta-dkms` (also 555) lets mangohud show up with `--mangoapp`, although the GPU stats are all 0. Maybe that's because of the beta driver?
  
About the HDR flags, the only issue I found with them is that some games will simply freeze after some seconds or minutes, but I think this bug has been reported already and it's still under investigation.

**Comment by Virkkunen** [comment]
> Found out that the `nvidia-open-beta-dkms` (driver 555) was the issue. Changing to `nvidia-beta-dkms` (also 555) lets mangohud show up with `--mangoapp`, although the GPU stats are all 0. Maybe that's because of the beta driver?
> 
> ![20240604_090945](https://private-user-images.githubusercontent.com/9111925/336341389-52f8a3aa-47b9-484f-85ae-de220e0e7517.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0ODUzNTMsIm5iZiI6MTcxNzQ4NTA1MywicGF0aCI6Ii85MTExOTI1LzMzNjM0MTM4OS01MmY4YTNhYS00N2I5LTQ4NGYtODVhZS1kZTIyMGUwZTc1MTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDRUMDcxMDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTYzYTQ3MTAzMjgyZWE4MTliN2U4NzQ1Yjg3ZGY2ZTE1MjVmN2RjOWI1YzNiZDRjY2JhMjE2MmRmMTY0N2ZjNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.DKPe5JxdgJxMfAl5ncvJJCHfH4ikbTg1bZMSqXV_dNc)
> 

Running the same games with just `mangohud %command%`, all stats are displayed without issue, only when running with gamescope and `--mangoapp` that the GPU stats aren't displayed. I also didn't find anything out of the ordinary on the console output while running with gamescope.

**Comment by chikobara** [comment]
I got this same warning message -but i solved it- while running on gnome wayland, using intel iris xe igpu, these flags i was using `gamescope -h 540 -H 1080 -f -F fsr --expose-wayland --mangoapp -- %command%`
i solved it by replacing `--expose-wayland` with `--backend wayland` and it worked

**Comment by K1NGSSTH** [comment]
I'm having this issue as well, i searched and tried many things that it could be related, i tried --backend wayland, --backend sdl, ENABLE_GAMESCOPE_WSI=0, ENABLE_GAMESCOPE_WSI=1, DISABLE_GAMESCOPE_WSI=0, DISABLE_GAMESCOPE_WSI=1, but i always get this error message, i also tried a package from AUR called gamescope-plus and with that package it works but with the latest gamescope or gamescope git it does not.

I test it with driver 570 and now with 575 beta nvidia driver, had the same outcome with bot drivers, also i believe is important to point out i on hyprland.

**Comment by Android789515** [comment]
Same issue. I simply use `gamescope -- %command%` and it fails:

```
CreateSwapchainKHR: Creating swapchain for non-Gamescope swapchain.
Hooking has failed somewhere!
You may have a bad Vulkan layer interfering.

Press OK to try to power through this error, or Cancel to stop.
```

## Edit:
This is only the case for certain games. Other games will work with gamescope but certain ones simply fail with the above error.

## Edit
Gemini said some users report pressing OK to "power through" the error *twice* gets the game to run. I tested and it seems to be the case, however, **I'm not sure what consequences** that has.
